scripts/ci/build.sh

```diff
@@ -0,0 +1,24 @@
+#!/usr/bin/env bash
+set -ex
+
+ROOTDIR="$(dirname "$0")/../.."
+cd "${ROOTDIR}"
+
+# shellcheck disable=SC2166
+if [ "$CIRCLE_BRANCH" = release -o -n "$CIRCLE_TAG" -o -n "$FORCE_RELEASE" ]; then echo -n >prerelease.txt; else date -u +"nightly.%Y.%-m.%-d" >prerelease.txt; fi
+if [ -n "$CIRCLE_SHA1" ]
+then
+  echo -n "$CIRCLE_SHA1" >commit_hash.txt
+fi
+
+mkdir -p build
+cd build
+
+# shellcheck disable=SC2166
+[ -n "$COVERAGE" -a "$CIRCLE_BRANCH" != release -a -z "$CIRCLE_TAG" ] && CMAKE_OPTIONS="$CMAKE_OPTIONS -DCOVERAGE=ON"
```

Comment:
I think that these `CIRCLE_` vars belong in the Circle CI config and the script should just be getting simple values via parameters instead (e.g. `NIGHTLY=0/1`, `COVERAGE=0/1`, `COMMIT_HASH`).

I was just looking at the config to see how `FORCE_RELEASE` and I forgot that I had this branch checked out. I have to say that having the outside variables magically accessible to the script makes it really hard to see how things are connected.

Reply:
Yes, we **should** change all this, but I'd *still* say we should just take exactly what's in the circleci config here and change it later in a second step ;-).
It's just a copy so far and it should stay like that for now...

If we're really bothered by that, we can just keep the circleci ``run_build`` stuff entirely over there and do the whole unification in a second step, but then we need to write another script for this here only to unify them later anyways...

Reply:
Well, it's not identical. It does have some changes, specifically to handle the `CIRCLE_` variables outside of CircleCI (e.g. the if for `$CIRCLE_SHA1`) so I thought this would be a change in the same category.

Anyway, I don't want to nitpick too much and bog down this PR with little tweaks so I'm fine with it staying as is, as long as it'll be refactored later.